### ISSUE TITLE
Stack overflow for equality testing with assumptions

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1692]
Imported from SourceForge on 2024-07-04 23:18:02
Created by **[tmonteil](https://sourceforge.net/u/tmonteil/)** on 2019-09-26 19:51:57
Original: https://sourceforge.net/p/maxima/bugs/3583

---

As reported on [this sagemath ticket](https://trac.sagemath.org/ticket/28538), we have:

~~~
(%i1) domain: complex;
(%o1)                               complex
(%i2) assume(x>0,y>0);
(%o2)                           [x > 0, y > 0]
(%i3) is(equal(y*(x-y),0));
~~~

leads to:

~~~
Maxima encountered a Lisp error:

 Condition in MACSYMA-TOP-LEVEL [or a callee]: INTERNAL-SIMPLE-ERROR: Bind stack overflow.

Automatically continuing.
To enable the Lisp debugger set *debugger-hook* to nil.
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:18:04
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2019-11-17 22:51:45
Original: https://sourceforge.net/p/maxima/bugs/3583/#3baa

---

- **labels**:  --> sign, domain, complex

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:18:07
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2019-11-17 22:51:45
Original: https://sourceforge.net/p/maxima/bugs/3583/#1b56

---

The bug appears to be in SIGNDIFF-SPECIAL in src/compar.lisp, which attempts to handle a special case, by constructing a new expression and studying the sign of that. The problem is that we end up in a loop -- in a few steps we end up back at an expression we tried already. 

I find that by suppressing that case when `domain` is `complex`, it prevents this bug. I've posted a message to the mailing list to see if anyone has an opinion about it. (The question is whether there is any useful case which is also suppressed.)

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:18:11
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2019-11-21 07:02:14
Original: https://sourceforge.net/p/maxima/bugs/3583/#0611

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 23:18:15
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2019-11-21 07:02:14
Original: https://sourceforge.net/p/maxima/bugs/3583/#7ef8

---

Fixed by commit 19ce22e (suppress special case in SIGNDIFF-SPECIAL when domain = complex).
